### PR TITLE
Latency measure

### PR DESCRIPTION
**Pull request by L0neGamer**
Add some utilities to measure latency

### CODE REVIEW

File: src/Discord/Internal/Gateway/EventLoop.hs [hunk 1]
```diff
@@ -182,11 +188,14 @@ runEventLoop thehandle sendablesData log = do loop
...
-                                        writeChan (librarySendables sendablesData) (Heartbeat sq)
+                                        sendHeartbeat sendablesData (gatewayHandleHeartbeatTimes thehandle) sq
                                         loop
       Right (InvalidSession retry) -> pure $ if retry then LoopReconnect else LoopStart
       Right Reconnect        -> pure LoopReconnect
-      Right HeartbeatAck     -> loop
+      Right HeartbeatAck     -> do
+        currTime <- getCurrentTime
+        result <- atomicModifyIORef' (gatewayHandleHeartbeatAckTimes thehandle) (dupe . const currTime)
+        result `seq` loop
```
Comment (yutotakano) [inline src/Discord/Internal/Gateway/EventLoop.hs:198]: Doesn't `atomicModifyIORef'` already force the result, so we don't need the `seq`? I'm going off the docs for `atomicallyModifyIORef'` here, please correct me if wrong.

Comment (L0neGamer) [reply src/Discord/Internal/Gateway/EventLoop.hs:198]: it seq's whatever is put into the ioref with what it outputs, so it needs this output to be evaluated as well

Comment (L0neGamer) [reply src/Discord/Internal/Gateway/EventLoop.hs:198]: This is an incorrect understanding, and I can adjust the functions as a result; let me know if there's anything else to change!

Comment (yutotakano) [reply src/Discord/Internal/Gateway/EventLoop.hs:198]: LGTM, but I can't do a local test for reasons, I trust you fully that it's fine though.

**Review by yutotakano — APPROVED** [review]

**Pull request merged**